org.alloytools.alloy.api/src/main/java/org/alloytools/alloy/core/api/TColumnType.java

```diff
@@ -0,0 +1,10 @@
+package org.alloytools.alloy.core.api;
+
+import java.util.List;
+
+/**
+ * The type of a column in a relation TField.
+ */
+public interface TColumnType {
```

Comment:
This type looks weird.  This type is a list of sigs, and a field type is a list of TColumnType; that means that a field type is `(tsig -> ... -> tsig) -> ... -> (tsig -> ... -> tsig)` instead of just `tsig -> ... -> tsig`.  How significant this type is to your implementation?  How difficult would it be to delete `TColumnType` and make a change as suggested above?

Reply:
I followed the implementation here. Daniel also referred to this. So I will follow you guys on this.